### DISCUSSION TITLE
Trying to get a handle on the internal data structures

### DISCUSSION THREAD

**Discussion by RonBOakes** [DraqueT/PolyGlot#1350] [Q&A]
I am presently working on a tool that will be made public after I use it for my Ph.D. dissertation.  I would like to provide the capability for users to get data out of their existing PolyGlot data and import it into my tool so that they can utilize the functions it will provide.

However, I have not been able to dig my way to find any indication of how either the XML (inside the ZIP files with .pgd extension) structures work in certain areas or how, once they are parsed in the Java code into the internal Java objects.  At this point, I have only been able to work out how to parse the XML file, extract the pronunciation rules, and convert them into the sound mapping rules for my tool.  I also understand the XML file Lexicon section reasonably enough to import it.  But I need to get my brain around the declension/conjugation sections.

For reference, my intention is to create a tool that will take a .pgd file and export it into a JSON object using the format documented [here](https://github.com/RonBOakes/conlang_json/blob/main/doc/conlang_json_spec.pdf).  I have a public open-source project for just this format (and a Python library providing some support).  This can be found at [https://github.com/RonBOakes/conlang_json](https://github.com/RonBOakes/conlang_json)

**Comment by DraqueT** [comment]
Heyo! I can definitely help you out there! Toss me an email over at draquemail (at) gmail (dot) com and we'll figure out a time to hash this out. Some of the data structures are a little gnarly. It would probably be easiest to do a web meeting or something for hgher bandwidth communication.
